theano/tensor/signal/pool.py

```diff
@@ -629,14 +637,15 @@ def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max'):
                 "Pool mode parameter only support 'max', 'sum',"
                 " 'average_inc_pad' and 'average_exc_pad'. Got %s" % mode)
         self.mode = mode
+        super(PoolGrad, self).__init__(openmp=openmp)
 
     def infer_shape(self, node, in_shapes):
         return [in_shapes[0]]
 
 
 class MaxPoolGrad(PoolGrad):
-    def __init__(self, ds, ignore_border, st=None, padding=(0, 0)):
-        PoolGrad.__init__(self, ds, ignore_border, st, padding, mode='max')
+    def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max', openmp=None):
+        PoolGrad.__init__(self, ds, ignore_border, st, padding, mode, openmp)
```

Comment:
Hi nouiz,

 def __init__(self, ds, ignore_border, st=None, padding=(0, 0), openmp=None):
    PoolGrad.__init__(self, ds, ignore_border, st, padding, mode='max', openmp)

    will trigger error

    " SyntaxError: non-keyword arg after keyword arg " (both in python2 and
python3)

    Try

    PoolGrad.__init__(self, ds, ignore_border, st, padding, openmp,
mode='max')

trigger

" TypeError: __init__() got multiple values for keyword argument 'mode' "


    Change the init of Pool_Grad to

 def __init__(self, ds, ignore_border, st=None, padding=(0, 0),
openmp=None, mode='max'):

 may pass the compilation , however since the sequence of parameter changes
the caller also has to change the sequence of arguments.


    I think the easiest workaround is by using mode  as the keyword
argument(mode='max') of the init and calling PoolGrad init with '... mode,
openmp'  and the current code that

calling PoolGrad need no changes. And the the caller of MaxPoolGrad also
need no changes since mode, openmp are appended.

2016-06-10 23:51 GMT+08:00 Frédéric Bastien <notifications@github.com>:

> In theano/tensor/signal/pool.py
> <https://github.com/Theano/Theano/pull/4607#discussion_r66636711>:
>
> >
> >      def infer_shape(self, node, in_shapes):
> >          return [in_shapes[0]]
> >
> >
> >  class MaxPoolGrad(PoolGrad):
> > -    def __init__(self, ds, ignore_border, st=None, padding=(0, 0)):
> > -        PoolGrad.__init__(self, ds, ignore_border, st, padding, mode='max')
> > +    def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max', openmp=None):
> > +        PoolGrad.__init__(self, ds, ignore_border, st, padding, mode, openmp)
>
> Use those 2 lines:
>
>  def __init__(self, ds, ignore_border, st=None, padding=(0, 0), openmp=None):
>     PoolGrad.__init__(self, ds, ignore_border, st, padding, mode, mode='max', openmp)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/Theano/pull/4607/files/00cc0fb836c3e834779ee5e0c9a45d9fb9edaeac#r66636711>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFdchva6vPKjXaB11_c9HL46zAJXqIA2ks5qKYgPgaJpZM4IxLqv>
> .
>